getting-started/src/main/kotlin/com/google/cloud/kotlin/demo/EmojifyController.kt

```diff
@@ -0,0 +1,152 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.kotlin.demo
+
+import com.google.cloud.storage.Bucket
+import com.google.cloud.storage.Storage
+import com.google.cloud.storage.StorageOptions
+import com.google.cloud.vision.v1.Likelihood
+import com.google.cloud.vision.v1.FaceAnnotation
+import com.google.cloud.vision.v1.AnnotateImageRequest
+import com.google.cloud.vision.v1.ImageAnnotatorClient
+import com.google.cloud.vision.v1.ImageSource
+import com.google.cloud.vision.v1.Image
+import com.google.cloud.vision.v1.Feature
+import com.google.cloud.vision.v1.Feature.Type
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.RequestParam
+import org.springframework.web.bind.annotation.RestController
+import java.awt.Polygon
+import java.io.FileOutputStream
+import java.io.PrintStream
+import java.nio.file.Files
+import java.nio.file.Paths
+import javax.imageio.ImageIO
+import java.util.logging.Logger
+
+val storage: Storage = StorageOptions.getDefaultInstance().service
+val bucket = storage.get(System.getenv("GOOGLE_STORAGE_BUCKET"))
+
+const val RESOURCES_PATH = "resources/" // Make sure to adapt path if running locally
+const val EMOJIS_PATH = RESOURCES_PATH + "emojis/"
+
+enum class Emoji {
+    JOY, ANGER, SURPRISE, SORROW, NONE
+}
+
+val emojiPic = mapOf(
+    Emoji.JOY to "joy.png",
+    Emoji.ANGER to "anger.png",
+    Emoji.SURPRISE to "surprise.png",
+    Emoji.SORROW to "sorrow.png",
+    Emoji.NONE to "none.png"
+)
+
+// Returns best emoji based on detected emotions likelihoods
+fun bestEmoji(annotation: FaceAnnotation): Emoji {
+    val emotionsLikelihood: Array<Likelihood> = arrayOf(Likelihood.VERY_LIKELY, Likelihood.LIKELY, Likelihood.POSSIBLE)
+    val emotions = mapOf(
+        Emoji.JOY to annotation.joyLikelihood,
+        Emoji.ANGER to annotation.angerLikelihood,
+        Emoji.SURPRISE to annotation.surpriseLikelihood,
+        Emoji.SORROW to annotation.sorrowLikelihood
+    )
+
+    for (likelihood in emotionsLikelihood) { // In this order: VERY_LIKELY, LIKELY, POSSIBLE
+        for (emotion in emotions) { // In this order: JOY, ANGER, SURPRISE, SORROW
+            if (emotion.value == likelihood) return emotion.key // Returns emotion corresponding to likelihood
+        }
+    }
+    return Emoji.NONE
+}
+
+class GcsObject(val bucketName: String, val blobName: String)
+class EmojifyResponse(val original: GcsObject, val emojified: GcsObject, val emojifiedUrl: String)
+
+@RestController
+class EmojifyController {
+    @GetMapping("/emojify")
+    fun emojify(@RequestParam(value = "objectName") objectName: String): EmojifyResponse? {
+
+        val publicUrl: String =
+            "https://storage.googleapis.com/${bucket.name}/emojified/emojified-$objectName" // api response
+        val log = Logger.getLogger(Application::class.java.name)
```

Comment:
logger is usually a private static final; this should be declared and initialized a class member, not inside of the mehtod.